src/ansys/fluent/core/utils/search.py

```diff
@@ -212,10 +238,320 @@ def inner(tree, path, root_path):
+    if similar_keys:
+        for key in similar_keys:
+            _search_wildcard(key)
+    else:
+        queries = _get_close_matches_for_word_from_names(search_string, names)
+        if queries:
+            _print_search_results(queries, wildcard=True)
+
+
+def search(
+    search_string: str,
+    language: Optional[str] = "eng",
+    wildcard: Optional[bool] = False,
+    match_whole_word: Optional[bool] = False,
+    match_case: Optional[bool] = False,
+):
+    """Search for a word through the Fluent's object hierarchy.
+
+    Parameters
+    ----------
+    search_string: str
+        The word to search for. Semantic search is default.
+    language: str
+        The language for the semantic search.
+        English is default for the semantic search.
+        ISO 639-3 code of the language to be used for semantic search.
+        See `https://omwn.org/omw1.html` for the list of supported languages.
+        The default value is `eng` for English language.
+    wildcard: bool
+        Whether to use wildcard pattern. If ``True`` will match wildcard pattern based on ``fnmatch`` module and
+        will turn off semantic matching.
+    match_whole_word: bool
+        Whether to get exact match. If ``True`` will match exact string and will turn off semantic matching.
+    match_case: bool
+        Whether to match case. If ``True`` will match case-insensitive case.
+
+    Raises
+    ------
+    ValueError
+        If both ``wildcard`` and ``match_whole_word`` are ``True``.
+
+    Examples
+    --------
+    >>> import ansys.fluent.core as pyfluent
+    >>> pyfluent.search("font", match_whole_word=True)
+    >>> pyfluent.search("Font")
+    >>> pyfluent.search("iter*", wildcard=True)
+    >>> pyfluent.search("读", language="cmn")   # search 'read' in Chinese
+    The most similar API objects are:
+    <solver_session>.file.read (Command)
+    <solver_session>.file.import_.read (Command)
+    <solver_session>.mesh.surface_mesh.read (Command)
+    <solver_session>.tui.display.display_states.read (Command)
+    <meshing_session>.tui.display.display_states.read (Command)
+    """
+    if wildcard and match_whole_word:
+        raise ValueError(
+            "``wildcard`` cannot be ``True`` when ``match_whole_word`` is ``True``."
+        )
+    elif language and match_whole_word:
+        warnings.warn(
+            "``match_whole_word=True`` will match exact string and will turn off semantic matching.",
+            UserWarning,
+        )
+    elif match_whole_word:
+        warnings.warn(
+            "``match_whole_word=True`` will turn off wildcard matching.",
+            UserWarning,
+        )
+
+    if wildcard:
+        _search_wildcard(search_string)
+    elif match_whole_word:
+        _search_whole_word(search_string, match_case)
+    else:
+        try:
+            _search_semantic(search_string, language)
+        except Exception:
+            _download_nltk_data()
```

Comment:
what is the exception if nltk data is missing? We should handle that exception specifically. This handling does not apply to other exceptions.

Reply:
@seanpearsonuk 

If package is not available on the website then it gives `ValueError`.

If package is not available locally then it gives `LookupErrror`.

We have explored these 2 exceptions that we have seen locally. 

So to handle unknown exceptions in future we have used `Exception`.

Reply:
I think catching just LookupError is better. As we know what which package will be downloaded, ValueError will not occur.

(Ideally, nltk should have provided custom error type, we could then catch that specific type)

Reply:
Ok. We will update this soon. Thank you.

Reply:
Done. Thank you.